src/client/resources/web/js/message_types.mjs

```diff
@@ -20,7 +23,7 @@
  * Chat message from Minecraft
  * @typedef {Object} ChatMessage
  * @property {'chatMessage'} type
- * @property {import('./message_parsing.mjs').Component} payload
+ * @property {{ ping: boolean, component: Component }} payload
```

Comment:
@creesch This is what I'm thinking. I don't feel that a separate metadata field is necessary. The component isn't special and shouldn't be considered the "default" field. It's just one of many fields in a JSON object.

Reply:
It is special in the sense that it now is specifically Minecrafts meesaging format. As soon as we start adding fields that is no longer the case.

But, I also don't know if I feel very strongly about it.

Reply:
I haven't added fields. The component field is still the official Minecraft format

Reply:
I shouldn't try to read this while on my phone

Reply:
I recommend the GitHub app. It's pretty good

Reply:
It's more that on the small screen I just tend to overlook things regardless of how well it is formatted. It is clear enough now that you are just expanding it be an object where component is a property.